api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/ProcessRdfForm.java

```diff
@@ -279,28 +283,40 @@ public static AdditionsAndRetractions addDependentDeletes( AdditionsAndRetractio
         return changes;
     }
 
-    public static void applyChangesToWriteModel(
-            AdditionsAndRetractions changes,
-            Model queryModel, Model writeModel, String editorUri) {
-        //side effect: modify the write model with the changes
-        Lock lock = null;
-        try{
-            lock =  writeModel.getLock();
-            lock.enterCriticalSection(Lock.WRITE);
-            if( writeModel instanceof OntModel){
-                ((OntModel)writeModel).getBaseModel().notifyEvent(new EditEvent(editorUri,true));
-            }
-            writeModel.add( changes.getAdditions() );
-            writeModel.remove( changes.getRetractions() );
-        }catch(Throwable t){
-            log.error("error adding edit change n3required model to in memory model \n"+ t.getMessage() );
-        }finally{
-            if( writeModel instanceof OntModel){
-                ((OntModel)writeModel).getBaseModel().notifyEvent(new EditEvent(editorUri,false));
-            }
-            lock.leaveCriticalSection();
-        }
+    public static void applyChangesToWriteModel(AdditionsAndRetractions changes, RDFService rdfService,
+    		String defaultGraphUri, String editorUri) {
+		appyChangesToWriteModel(changes, editorUri, rdfService, defaultGraphUri);
     }
+    
+	@Deprecated
+	public static void applyChangesToWriteModel(AdditionsAndRetractions changes, Model writeModel, String editorUri) {
+		RDFService rdfService = new RDFServiceModel(writeModel);
+		appyChangesToWriteModel(changes, editorUri, rdfService, null);
+	}
+
+	private static void appyChangesToWriteModel(AdditionsAndRetractions changes, String editorUri,
+			RDFService rdfService, String graphUri) {
+		ChangeSet cs = rdfService.manufactureChangeSet();
+		cs.addPreChangeEvent(new BulkUpdateEvent(null, true));
+		cs.addPostChangeEvent(new BulkUpdateEvent(null, false));
+		
+		ByteArrayOutputStream additionsStream = new ByteArrayOutputStream();
+		changes.getAdditions().write(additionsStream, "N3");
+		InputStream additionsInputStream = new ByteArrayInputStream(additionsStream.toByteArray());
+		
+		ByteArrayOutputStream retractionsStream = new ByteArrayOutputStream();
+		changes.getRetractions().write(retractionsStream, "N3");
+		InputStream retractionsInputStream = new ByteArrayInputStream(retractionsStream.toByteArray());
+		
+		cs.addAddition(additionsInputStream, RDFServiceUtils.getSerializationFormatFromJenaString("N3"), graphUri, editorUri);
+		cs.addRemoval(retractionsInputStream, RDFServiceUtils.getSerializationFormatFromJenaString("N3"), graphUri, editorUri);
+
+        try {
+			rdfService.changeSetUpdate(cs);
+		} catch (RDFServiceException e) {
+			log.error(e, e);
+		}
+	}
```

Comment:
Not sure I understand the reason for existence (decoupling) of the first and third methods in this part. Can you please double check can we merge that into one method? Also, I am not sure it is a good practice having methods only different in order of parameters.

Reply:
Should be fixed now

Reply:
I removed deprecated method as it is not used anywhere in the code and creation RDFService from model is not considered as a good practice.